src/Httpstatus.php

```diff
@@ -348,15 +359,16 @@ public function hasReasonPhrase($statusText)
      */
     public function getResponseClass($statusCode)
     {
-        $statusCode = $this->filterHttpStatusCode($statusCode);
-        $firstDigit = (int) substr($statusCode, 0, 1);
+        $responseClass = [
+            1 => self::CLASS_INFORMATIONAL,
+            2 => self::CLASS_SUCCESS,
+            3 => self::CLASS_REDIRECTION,
+            4 => self::CLASS_CLIENT_ERROR,
+            5 => self::CLASS_SERVER_ERROR,
+        ];
 
-        switch ($firstDigit) {
-            case 1: return self::CLASS_INFORMATIONAL;
-            case 2: return self::CLASS_SUCCESS;
-            case 3: return self::CLASS_REDIRECTION;
-            case 4: return self::CLASS_CLIENT_ERROR;
-            case 5: return self::CLASS_SERVER_ERROR;
-        }
+        $statusCode = $this->filterStatusCode($statusCode);
+
+        return $responseClass[floor($statusCode / 100)];
```

Comment:
Maybe, just this one, as it is easier to read/understand with the `substr. That would be nice. Than I will merge this.